### ISSUE TITLE
Minor Gameplay style overhaul idea

### ISSUE THREAD

**Issue by LiamVallance** [Hodge0699/McompProject#63]
This goes along with the smaller rooms idea a little as i was thinking of maybe having it so that the camera does not follow the player but just shows an overview of the entire room, i also was wondering what people thought of a sort of fog of war / line of sight style mechanic where the player cannot see everything around him and can only see close to him and within a vision cone ahead of him, this could make the game a little more tense, put more value in using the time mechanics (e.g. rewind if an enemy comes out of nowhere or being able to look around the room while everything is slow-motion for information on enemy positions) and would allow for smaller enemy counts and smaller rooms while keeping the game fast paced and interesting. What are people opinions on this, if you like it i am happy to look into the vision system myself.
![player vision cone](https://user-images.githubusercontent.com/10143803/51768317-de3b5a80-20d7-11e9-97f9-143944b9bc94.png)

**Comment by LiamVallance** [comment]
Any obstacles in the level (if we added some) could also block players line of sight too again meaning an enemy could jump out at you.

**Comment by 14016036** [comment]
I personally don’t think this should be considered “minor” as it does affect a lot of the components already in the game...

And while I really like the idea I’m not 100% sure it goes with a gungeon style game like we currently have, while I think it would force players to utilise the time mechanics more, I still feel they would play a hell of a lot slower and cautious potentially using the time mechanics just as little as you can now. I also think this “gameplay” would clash a lot with the time mechanics like slow down/speed up.... but I think it really enhances the rewind time mechanic.

So while I’m typing this I kinda like the idea of us somehow “lore wise” have this mechanic on the floor with that rewind time... maybe if we could make some interactive slides prior to the level being loaded explaining that the air vents are broken and it’s completely smoked out this floor..

Or something similar 

But this is pure theory as I don’t know how people would respond to it.

**Comment by LiamVallance** [comment]
How about we go with this smoked out level 2 for now and see how it plays, if it feels horrible then we can just leave it out but with the mechanic implemented we could easily trial it on the other levels and see how it plays. On another idea path it could always be a difficulty option, so the hardest difficulty has the vision cone but for those who don't like it they could just player one difficulty down.

**Comment by JakeP121** [comment]
I agree it kinda goes against the whole bullet hell theme if we hide the level from the player but then again it could be one of those things we could write about in the report saying why it does/doesn’t work. 

I have had a go at implementing fog of war over the summer and it is pretty difficult though, and including the line of sight in it would be even harder. 

I do like the idea we had a bit ago of having the minimap show the entire level but only render the enemies that are in view.
